Carta.py

```diff
@@ -320,33 +341,73 @@ def opponentRespondsInGrabPhase(self):
         # Both players grabbed the right card, but you grabbed it faster than opponent
         if ((self.GPinfo.yourGrabCardLastWord == self.curReadingCard.getLastWord()) and \
             (self.GPinfo.oppoGrabCardLastWord == self.curReadingCard.getLastWord()) and \
             (self.GPinfo.yourTime < self.GPinfo.opponentTime)):
             self.GPinfo.youWin = True
             self.GPinfo.opponentWin = False
             statement = "Carta.py: decideWinner: Both players grabbed the right card, but you grabbed it faster than opponent"
+            self.dialogs.append("You Win, Opponent Loses")
+            if (self.GPinfo.yourGrabCardSide == GrabbingCardSide.YOUR_SIDE):
+                self.numYourGrabCardInPlay -= 1
+            elif (self.GPinfo.yourGrabCardSide == GrabbingCardSide.OPPONENT_SIDE):
+                self.numOppoGrabCardInPlay -= 1
+                self.phase = Phase.YOUR_ADJUSTMENT
+                self.numCardsToTransfer += 1
         # Both players grabbed the right card, but opponent grabbed it faster than you
         elif ((self.GPinfo.yourGrabCardLastWord == self.curReadingCard.getLastWord()) and \
               (self.GPinfo.oppoGrabCardLastWord == self.curReadingCard.getLastWord()) and \
               (self.GPinfo.yourTime > self.GPinfo.opponentTime)):
             self.GPinfo.youWin = False
             self.GPinfo.opponentWin = True
             statement = "Carta.py: decideWinner: Both players grabbed the right card, but opponent grabbed it faster than you"
-        # you didn't grab the right card, but opponent grabbed the right card
+            self.dialogs.append("Opponent Wins, You Lose")
+            self.shouldOpponentTransfer(False)
+        # you grabbed the wrong card, but opponent grabbed the right card
         elif ((self.GPinfo.yourGrabCardLastWord != self.curReadingCard.getLastWord()) and \
               (self.GPinfo.oppoGrabCardLastWord == self.curReadingCard.getLastWord())):
             self.GPinfo.youWin = False
-            statement = "Carta.py: decideWinner: you didn't grab the right card, but opponent grabbed the right card"
+            self.GPinfo.opponentWin = True
+            statement = "Carta.py: decideWinner: you grabbed the wrong card, but opponent grabbed the right card"
+            self.dialogs.append("Opponent Wins, You Lose")
+            self.shouldOpponentTransfer(True)
         # you grabbed the right card, but opponent didn't grab the right card
         elif ((self.GPinfo.yourGrabCardLastWord == self.curReadingCard.getLastWord()) and \
               (self.GPinfo.oppoGrabCardLastWord != self.curReadingCard.getLastWord())):
             self.GPinfo.youWin = True
             self.GPinfo.opponentWin = False
             statement = "Carta.py: decideWinner: you grabbed the right card, but opponent didn't grab the right card"
+            self.dialogs.append("You Win , Opponent Loses")
+            if (self.GPinfo.yourGrabCardSide == GrabbingCardSide.YOUR_SIDE):
```

Comment:
`GrabbingCardSide` is not needed

`(self.GPinfo.yourGrabCard.getStatus() == GrabCardStatus,YOU)`